src/core/distributions/phylogenetics/substitution/AbstractPhyloCTMCSiteHomogeneous.h

```diff
@@ -4090,11 +4085,13 @@ void RevBayesCore::AbstractPhyloCTMCSiteHomogeneous<charType>::swapParameterInte
 template<class charType>
 void RevBayesCore::AbstractPhyloCTMCSiteHomogeneous<charType>::touchSpecialization( const DagNode* affecter, bool touch_all )
 {
-
     if ( touched == false )
     {
         touched = true;
         this->storedLnProb = this->lnProb;
+
+        prev_dirty_nodes = dirty_nodes;
+        prev_pmat_dirty_nodes = pmat_dirty_nodes;
```

Comment:
When we touch this distribution, save copies of the lnProb, the dirty_nodes vector, and the pmat_dirty_nodes vector, so we can restore them later.